lib/core-common/src/utils/normalize-stories.ts

```diff
@@ -24,10 +27,17 @@ export const normalizeStoriesEntry = (
   if (typeof entry === 'string') {
     if (!entry.includes('**') && isDirectory(configDir, entry)) {
       directory = entry;
-      titlePrefix = DEFAULT_TITLE_PREFIX;
       files = DEFAULT_FILES;
+      titlePrefix = DEFAULT_TITLE_PREFIX;
     } else {
-      glob = entry;
+      const match = entry.match(GLOB_REGEX);
+      if (match) {
+        directory = match.groups.directory;
+        files = match.groups.files;
+        titlePrefix = DEFAULT_TITLE_PREFIX;
```

Comment:
@shilman I added this to support common glob cases. This feels powerful for backwards compatibility vs forcing folks to use the directory/files config but i guess we could do with out it. Thoughts?

Reply:
@tmeasday WDYT? I didn't add this in the first place due to "too much magic", but it doesn't seem too bad 🤷‍♂️

Reply:
I guess it depends on what we want the end state to be. Without this change folks will have to upgrade to the simple version without files extensions. However, that forces users of server to use the directory/files version vs a regex (there is no way to configure default files)

Reply:
I think this makes sense, although I am not 100% clear on the implications. Does it just mean if you are using the `dir/**/*.stories.js` type syntax you can now leave titles off components where you couldn't before?

That seems like a good thing to me.